### PR TITLE
Safely add distinct on preload_lateral

### PR DESCRIPTION
**Pull request by cspeper**
Only add a distinct to a query if it does not already have one to avoid:
```
%Ecto.Query.CompileError{message: "only one distinct expression is allowed in query"
```

### CODE REVIEW

File: lib/dataloader/ecto.ex
Comment (peaceful-james) [inline lib/dataloader/ecto.ex:897]: Would this be cleaner if it was pattern-matched in the function arguments?
For example
```
defp maybe_distinct(%Ecto.Query{distinct: dist} = query, _) when dist, do: query
```
Then the rest of the patterns can assume `dist` is false or nil.
```
defp maybe_distinct(query, [%Ecto.Association.Has{}, %Ecto.Association.BelongsTo{} | _]), do: distinct(query, true)

defp maybe_distinct(query, [%Ecto.Association.ManyToMany{} | _]), do: distinct(query, true)
```

File: test/dataloader/ecto/limit_query_test.exs
Comment (benwilson512) [inline test/dataloader/ecto/limit_query_test.exs:46]: Can we make a dedicated clause for this? With this change every clause is distinct, which would make test cases showing that we _add_ a distinct no longer valid.

**Review by benwilson512 — CHANGES_REQUESTED** [review]
Thanks for the PR! A small change to the test and we can get this merged in.

**Comment by cspeper** [comment]
@benwilson512 @peaceful-james Sorry for the delay and thanks for the feedback!  I think I addressed both comments.

**Review by benwilson512 — APPROVED** [review]
Great, thank you!

**Comment by benwilson512** [comment]
@cspeper can you run `mix format` on the code base so that the CI passes?

**Comment by cspeper** [comment]
@benwilson512 Looks like the format issue was on master but I went ahead and ran the update.  I'll merge once it's approved and passes CI.  Thanks!

**Comment by benwilson512** [comment]
Ahhh haha `then` isn't supported on older Elixirs. I'll make a tweak to replace it with a regular function.

**Comment by cspeper** [comment]
> Ahhh haha `then` isn't supported on older Elixirs. I'll make a tweak to replace it with a regular function.



> Ahhh haha `then` isn't supported on older Elixirs. I'll make a tweak to replace it with a regular function.

Hi @benwilson512 Is there anything more you needed me to do with this PR?  It looks like master has the same failure

**Comment by benwilson512** [comment]
Ah good point, oK we're good to go then, thanks!

**Pull request merged**
